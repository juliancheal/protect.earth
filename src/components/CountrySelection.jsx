import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';

import Countries from '../countries';

/**
 * If the user has a country preference in session storage, we don't need to
 * present the user to select a country. We render a button to _clear_ their selection.
 */
const CountryButton = ({}) => {
  // Get the country code. e.g., "BG" for "Bulgaria"
  const [country] = useState(sessionStorage.getItem('country_pref'));

  // Default to Great Britain
  const countryName = Countries.fromAlpha2Code(country || 'GB').name;
  const countryEmoji = Countries.fromAlpha2Code(country || 'GB').emoji;

  return (
    <div className="change-country">
      {countryName && countryEmoji ? (
        <>
          <span className="current selected-item">
            {countryName} {countryEmoji}
          </span>{' '}
        </>
      ) : null}

      <Button href="/select-your-country" className="link text-white">
        {countryName ? 'Change' : 'Filter site for your'} country
      </Button>
    </div>
  );
};

export default CountryButton;
