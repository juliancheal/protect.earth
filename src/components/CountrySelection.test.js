import React from "react";
import renderer from "react-test-renderer";

import CountrySelection from "./CountrySelection";

describe("CountrySelection", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<CountrySelection />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
