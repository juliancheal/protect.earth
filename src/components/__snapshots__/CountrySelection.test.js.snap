// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelection renders correctly 1`] = `
<div
  className="change-country"
>
  <span
    className="current selected-item"
  >
    United Kingdom
     
    ðŸ‡¬ðŸ‡§
  </span>
   
  <a
    className="link text-white btn btn-primary"
    href="/select-your-country"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Change
     country
  </a>
</div>
`;
